### ISSUE TITLE
Lua Binding: setPos doesn't handle tables correctly

### ISSUE THREAD

**Issue by zach2good** [project-topaz/topaz#1329]
**_I have:_**

- [x] searched existing issues (http://project-topaz.com/issues/) to see if the issue has already been opened
- [x] checked the commit log to see if the issue has been resolved since my server was last updated

**_Additional Information_** (Steps to reproduce/Expected behavior) **:** 

While I was working on the TOAU44 fix, I was using `instance:getEntryPos()` which gives back a table in the following form:
```lua
{
    x = xpos,
    y = ypos,
    z = zpos,
    r = rot,
}
```

setPos SHOULD take that as shown here:
https://github.com/project-topaz/topaz/blob/release/src/map/lua/lua_baseentity.cpp#L2732-L2752

But it was putting me through the floor.

I ended up using `setPos(table.x, table.y, table.z, table.r)` to skirt around the problem, but that isn't ideal

**Comment by Kreidos** [comment]
Looks like setPos only expects a plain list, or "numerically keyed" table. ie `{ xpos, ypos, zpos, rot }` and not the string-keyed tables which getEntryPos provides.

**Comment by TeoTwawki** [comment]
lua_rawgeti spaghetti did it

**Comment by TeoTwawki** [comment]
Should be fixable using 
```cpp
        lua_getfield(L, 1, "x");
        posX = (float)lua_tonumber(L, -1);
        lua_getfield(L, 1, "y");
        posY = (float)lua_tonumber(L, -1);
        lua_getfield(L, 1, "z");
        posZ = (float)lua_tonumber(L, -1);
```
instead

**Comment by Kreidos** [comment]
I already rewrote a version of setPos to handle either numerical or string keys, however, in-so-doing it uncovered some problems with how pathfind.lua functions are written. (Since they are passed to setPos) The old lua_rawgeti didn't reveal these issues.

Here was my version if anyone feels like looking into it before I have time:

```c++
        lua_pushnil(L);
        while (lua_next(L, 1))
        {
            char key = lua_tostring(L, -2)[0];
            switch (key)
            {
                case '1':
                case 'x':
                    m_PBaseEntity->loc.p.x = (float)lua_tonumber(L, -1);
                    break;
                case '2':
                case 'y':
                    m_PBaseEntity->loc.p.y = (float)lua_tonumber(L, -1);
                    break;
                case '3':
                case 'z':
                    m_PBaseEntity->loc.p.z = (float)lua_tonumber(L, -1);
                    break;
                case '4':
                case 'r':
                    m_PBaseEntity->loc.p.rotation = (uint8)lua_tointeger(L, -1);
                    break;
            }
            lua_pop(L, 1);
        }
```

**Comment by TeoTwawki** [comment]
yeah our path code has been failing to accept float values from our lua tables for a long while. https://github.com/project-topaz/topaz/commit/06758c41a7a94d4f51fbc8fff47c09be832cee51#diff-0af40b103e752b39d72d353b4907058e

(and instead of looking into why and fixing it, lazy Teo just gave Guivre a "good enough" non floating point set of coordinates)

**Comment by Kreidos** [comment]
From what I see, the cause may be a misunderstanding of how Lua tables are passed as return values. They're passed by reference (as pointer to the table) and the table's variable and location are re-used between calls. We should be returning direct references to the parent table, not building new tables, as those intermediate tables are being modified right from underneath us by subsequent calls to the pathing functions..
